crates/core/tedge_agent/src/http_server/entity_store.rs

```diff
@@ -170,35 +188,141 @@ async fn register_entity(
     ))
 }
 
-async fn patch_entity(
+async fn get_resource(
     State(state): State<AgentState>,
     Path(path): Path<String>,
-    Json(twin_fragments): Json<Map<String, Value>>,
 ) -> impl IntoResponse {
-    let topic_id = EntityTopicId::from_str(&path)?;
-    let twin_data = EntityTwinData::try_new(topic_id, twin_fragments)?;
+    let (topic_id, channel) = parse_path(&path)?;
+    match channel {
+        Channel::EntityMetadata => Ok(get_entity(state, topic_id).await.into_response()),
+        Channel::EntityTwinData { fragment_key } => {
+            if fragment_key.is_empty() {
+                return Ok(get_entity_twin_fragments(state, topic_id)
+                    .await
+                    .into_response());
+            }
 
-    let response = state
-        .entity_store_handle
-        .clone()
-        .await_response(EntityStoreRequest::Patch(twin_data))
-        .await?;
-    let EntityStoreResponse::Patch(res) = response else {
-        return Err(Error::InvalidEntityStoreResponse);
-    };
-    res?;
+            Ok(
+                get_entity_twin_fragment(state, topic_id, fragment_key.to_string())
+                    .await
+                    .into_response(),
+            )
+        }
+        _ => Err(Error::MethodNotAllowed),
+    }
+}
 
-    let entity = get_entity(State(state), Path(path)).await?;
+async fn put_resource(
+    State(state): State<AgentState>,
+    Path(path): Path<String>,
+    payload: String,
+) -> impl IntoResponse {
+    if payload.len() > HTTP_MAX_PAYLOAD_SIZE {
+        return Err(Error::EntityStoreError(
+            entity_store::Error::PayloadSizeExceeded(payload.len()),
+        ));
+    }
```

Comment:
Wasted a lot of time trying to use the [RequestBodyLimitLayer](https://docs.rs/tower-http/0.6.2/tower_http/limit/struct.RequestBodyLimitLayer.html) with axum `Router`, but failed. So, falling back to this instead.

Reply:
Well... It was my bad. I was wrongly trying to integrate the `tower-http` abstraction [RequestBodyLimitLayer](https://docs.rs/tower-http/0.6.2/tower_http/limit/struct.RequestBodyLimitLayer.html) instead [DefaultBodyLimit](https://docs.rs/axum/latest/axum/extract/struct.DefaultBodyLimit.html) which is the right `axum` router level abstraction. Resolved by 69a6f6f.